### ISSUE TITLE
"read operation took too long"

### ISSUE THREAD

**Issue by zong0** [keybase/client#2901]
Read timeout in /keybase/private..., asked by app to send a log so here it is, but also during 'keybase log send':

```
Continue sending logs to keybase.io? (type 'YES' to confirm): YES
▶ WARNING error opening log "/Users/[username]/Library/Logs/keybase.updater.log": open /Users/[username]/Library/Logs/keybase.updater.log: no such file or directory
```

my log id: 615bf1180e4369dfda13e51c

**Comment by strib** [comment]
Hi @zong0, thanks for the report.  It looks like you are unable to connect to one type of our KBFS backend servers (the block server, at bserver.kbfs.keybase.io).  Our server shows the last time you connected was on 5/22 at 1:20 pm PT.  Interestingly, possibly around that same time, it looks like the year on your device's clock was set to 2023:

```
2023-05-22T17:08:17.576146 ▶ [DEBU kbfs connection.go:316] 266c (CONN BlockServerRemote cc62ba32) Connection: dialing transport
2023-05-22T17:08:17.615280 ▶ [WARN kbfs connection.go:321] 266d (CONN BlockServerRemote cc62ba32) Connection: error dialing transport: x509.CertificateInvalidError{Cert:(*x509.Certificate)(0xc828122900), Reason:1}
2023-05-22T17:08:17.615332 ▶ [WARN kbfs(BSR) connection.go:486] 266e connection error: x509: certificate has expired or is not yet valid; retrying in 971.047419ms
...
2023-05-22T17:08:45.414385 ▶ [WARN kbfs connection.go:321] 2682 (CONN BlockServerRemote cc62ba32) Connection: error dialing transport: x509.CertificateInvalidError{Cert:(*x509.Certificate)(0xc8201b0000), Reason:1}
2023-05-22T17:08:45.414459 ▶ [WARN kbfs(BSR) connection.go:486] 2683 connection error: x509: certificate has expired or is not yet valid; retrying in 10.904593293s
2016-05-22T17:09:09.544864 ▶ [DEBU FUSE serve.go:776] 2684 <- Statfs [ID=0x4 Node=0x1 Uid=501 Gid=20 Pid=1770]
```

To you, it looked like our block server cert had expired.  You can see on the last line above that the clock went back to 2016 -- but I wonder if we have a bug where our retry timer is still waiting for the clock to reach 17:08:55 in 2023 before it tried to reconnect to the block server again.  I thought our timer would be based on the system monotonic clock, but I will have to double check that.

I will look into it, but in the meantime it seems like restarting KBFS might fix things for you.

**Comment by zong0** [comment]
Yes, I did set my clock to 2023 to test some (other unrelated) cert, and set it back afterwards. Keybase may have been open at that time (under a different logged in user, no less).

Restarting unfortunately did not help.

**Comment by strib** [comment]
Hrm, strange that restarting didn't help.  Would you mind do another `keybase log send`, post-restart, and letting us know the id?  Maybe something else is going on.

**Comment by zong0** [comment]
Sure, new log id: 4aa49935efab98b01dbfe01c

**Comment by zong0** [comment]
Update: The process kbfs was hanging around and pegging the CPU just now all of a sudden (the UI app was quit an hour earlier, does kbfs usually hang around?). Anyway, I force quit the process kbfs and after opening the app again to check, now I don't get this error any more.

**Comment by strib** [comment]
Ah ok great, glad it's working now.  What did you do to restart exactly?  Just quit from the UI?  That only kills the UI -- to restart kbfs you need to `keybase launchd restart keybase.kbfs`.

Not sure why it was pegging the CPU -- looks like maybe program on your device (a file indexer?) was in a loop trying to access files.

We are still looking into the timer issue.

**Comment by zong0** [comment]
Yes, I previously only quit from the UI and also did `keybase ctl restart`, didn't know to do `keybase launchd restart keybase.kbfs`. Thanks.

**Comment by BenjaminPoilve** [comment]
Just for information, ran into the same issue on OSX (log ID: ae4f2dfaf5a8dee8af7b8e1c). A restart later (with keybase launchd restart keybase.kbfs) and everything went back to normal.
